src/test/java/org/htmlunit/javascript/host/URLSearchParamsTest.java

```diff
@@ -58,6 +58,181 @@ public void ctor() throws Exception {
         loadPageVerifyTitle2(html);
     }
 
+    @Test
+    @Alerts(DEFAULT = {"", "Failed to construct 'URLSearchParams': The provided value cannot be converted to a sequence.",
+            "foo=1&bar=2", "Failed to construct 'URLSearchParams': Sequence initializer must only contain pair elements.",
+            "1%2C2=1%2C2%2C3", "Failed to construct 'URLSearchParams': The object must have a callable @@iterator property."},
+            IE = {})
+    public void ctorArray() throws Exception {
+        final String html =
+            "<html>\n"
+            + "<head>\n"
+            + "  <script>\n"
+            + LOG_TITLE_FUNCTION
+            + "    function test() {\n"
+            + "      if (self.URLSearchParams) {\n"
+            + "        log(new URLSearchParams(Array(0)));\n"
+            + "        try {\n"
+            + "          log(new URLSearchParams(Array(1)));\n"
+            + "        } catch (ex) {\n"
+            + "          log(ex.message);\n"
+            + "        }\n"
+            + "        log(new URLSearchParams([\n"
+            + "          ['foo', '1'],\n"
+            + "          ['bar', '2'],\n"
+            + "        ]));\n"
+            + "\n"
+            + "        try {\n"
+            + "          log(new URLSearchParams([\n"

```

Comment:
it's slightly misleading to have this
```suggestion
            + "          new URLSearchParams([\n"

```